lib/msf/ui/console/command_dispatcher/payload.rb

```diff
@@ -58,7 +58,7 @@ def cmd_to_handler(*_args)
               }
             }
 
-            handler.datastore.merge!(mod.datastore)
+            handler.share_datastore(mod.datastore)
```

Comment:
multi/handler datastore:
```ruby
> handler.datastore
=> {
   ...
 "lhost"=>"192.168.123.1",
 "lport"=>"80",
  ...
}
```

Payload datastore:
```ruby
> mod.datastore
=> {
...
 "LHOST"=>nil,
 "LPORT"=>"4444",
....
}
```

Result of old `handler.datastore.merge!(mod.datastore)` doesn't merge with case sensitivity taken into consideration:
```json
> handler.datastore.merge(mod.datastore)
=> {
...  
 "lhost"=>"192.168.123.1",
 "lport"=>"80",
   ...
 "LHOST"=>nil,
 "LPORT"=>"4444",
}
```

With this change they are correctly merged together:
```ruby
> handler.datastore
=> {
...
 "LHOST"=>nil,
 "LPORT"=>"4444",
...
}
```

Reply:
Note that LHOST being nil is intended, as it will default to my setg value:
```
> setg

Global
======

  Name      Value
  ----      -----
  lhost     192.168.123.1
  loglevel  3

```

Reply:
This suggests this issue exists elsewhere too. Potentially worth an investigation.